File: src/6-kyu/does_my_number_look_big_in_this/python/main.py
import math
from functools import reduce

def narcissistic(value):
    length = int(math.log10(value)) + 1
    def fold_fn(acc, _):
        n, total = acc
        digit = n % 10
        return (n // 10, total + digit ** length)
    return reduce(fold_fn, range(length), (value, 0))[1] == value

File: src/6-kyu/does_my_number_look_big_in_this/typescript/main.ts
export function narcissistic(value: number, n = value,
	numberDigits = Math.floor(Math.log10(value)) + 1, sum = 0): boolean {
	return n === 0 ? sum === value : narcissistic(value, Math.floor(n / 10),
		numberDigits, sum + ((n % 10) ** numberDigits));
}
